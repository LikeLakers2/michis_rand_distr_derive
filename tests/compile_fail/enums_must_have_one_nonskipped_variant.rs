use michis_rand_distr_derive::StandardDistribution;
fn main() {}

#[derive(StandardDistribution)]
pub enum ErrorsIfEnumHasNoNonSkippedVariants1 {
	#[standard_distribution(skip)]
	Variant,
}

#[derive(StandardDistribution)]
pub enum ErrorsIfEnumHasNoNonSkippedVariants2 {
	#[standard_distribution(skip)]
	Variant1,
	#[standard_distribution(skip)]
	Variant2,
}
