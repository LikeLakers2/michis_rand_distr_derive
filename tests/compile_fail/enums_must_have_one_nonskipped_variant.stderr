error: There must be at least one non-skipped variant
 --> tests/compile_fail/enums_must_have_one_nonskipped_variant.rs:4:10
  |
4 | #[derive(StandardDistribution)]
  |          ^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the derive macro `StandardDistribution` (in Nightly builds, run with -Z macro-backtrace for more info)

error: There must be at least one non-skipped variant
  --> tests/compile_fail/enums_must_have_one_nonskipped_variant.rs:10:10
   |
10 | #[derive(StandardDistribution)]
   |          ^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the derive macro `StandardDistribution` (in Nightly builds, run with -Z macro-backtrace for more info)
